app/controllers/api/databases.php

```diff
@@ -1869,43 +1865,35 @@ function createAttribute(string $databaseId, string $collectionId, Document $att
             }
         }
 
-        // Check collection permissions when enforced
-        if ($collection->getAttribute('permission') === 'collection') {
-            $validator = new Authorization('write');
-            if (!$validator->isValid($collection->getWrite())) {
-                throw new Exception('Unauthorized permissions', 401, Exception::USER_UNAUTHORIZED);
-            }
+        $validator = new Authorization('create');
+        if (!$validator->isValid($collection->getCreate())) {
+            throw new Exception('Unauthorized permissions', 401, Exception::USER_UNAUTHORIZED);
         }
 
-        $data['$collection'] = $collection->getId(); // Adding this param to make API easier for developers
-        $data['$id'] = $documentId == 'unique()' ? $dbForProject->getId() : $documentId;
-        $data['$read'] = (is_null($read) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $read ?? []; //  By default set read permissions for user
-        $data['$write'] = (is_null($write) && !$user->isEmpty()) ? ['user:' . $user->getId()] : $write ?? []; //  By default set write permissions for user
-
-        // Users can only add their roles to documents, API keys and Admin users can add any
-        $roles = Authorization::getRoles();
+        $permissions = PermissionsProcessor::addDefaultsIfNeeded(
```

Comment:
Not sure I fully understand everything that goes here, seems like a lot of added complexity that I think we could try and reduce with abstraction and interfaces that explain the logic better.

Reply:
Could this potentially be part of the validation logic. Would love some description in a docblock.

Reply:
This is adding the permissions for the current user for any that are missing on a document (and file) creation

E.g. if a user creates a document with only `"read("any")"` permissions, this will add "update("user:xxx")" and "delete("user:xxx")", this was existing logic, is it expected with the new model?